GenericModConfigMenu/Api.cs

```diff
@@ -52,41 +50,51 @@ public interface IApi
 
     public class Api : IApi
     {
+        /*********
+        ** Public methods
+        *********/
         public void RegisterModConfig(IManifest mod, Action revertToDefault, Action saveToFile)
         {
+            this.AssertNotNull(mod, nameof(mod));
+            this.AssertNotNull(revertToDefault, nameof(revertToDefault));
+            this.AssertNotNull(saveToFile, nameof(saveToFile));
```

Comment:
Is  there enough information to tell us which mod invoked `RegisterModConfig` with null args?

The three `if`s I added was meant to do that so we can pinpoint which mod invoked this method in a wrong way.

Reply:
Yep. Since we're throwing an exception, SMAPI should intercept and log it under the mod calling the method. So we'd see something like:
```
[Some Broken Mod] This mod failed in the GameLoop.GameLaunched event. Technical details:
System.ArgumentNullException: Value cannot be null. Parameter name: revertToDefault.
at ...
```
Though I'd be fine adding the `IManifest` as a optional parameter in the null checks; if the manifest itself isn't null we can add the mod name to the exception message.

Reply:
Awesome! Okay then, I have no more concerns. LGTM! 🚀